packages/client/src/client/open-feature-client.ts

```diff
@@ -49,6 +49,10 @@ export class OpenFeatureClient implements Client {
     };
   }
 
+  get providerStatus(): ProviderStatus {
+    return this.providerAccessor()?.status || ProviderStatus.READY;
+  }
+
```

Comment:
This is the only change outside react. It makes it possible to get he associated provider state immediately, which prevents us from doing some extra work in cases when the provider is already ready.

We could do this in a separate PR if we don't want to pollute the web-sdk changelog with a "feat" that doesn't really apply to it.

Reply:
Should we add this to the spec? It may be useful on the server too.

What would you label it if you didn't go with feature?

Reply:
> Should we add this to the spec? It may be useful on the server too.

I expected somebody to ask this question, and I don't feel too strongly about it. Honestly I've considered adding it before, but even with all the previous event stuff, this is the first time I've needed it. Since handlers run immediately, you can always just add one and get the state, and in most cases that's the better choice because it will cover the case where the provider is _about_ to become ready as well. In fact we might want to discourage it's use generally, to avoid confusing race conditions.

The only reason is really helpful here is because, in the case of UIs, the page is refreshed and components mount/unmount a lot. That would mean a lot of handler additions/removals. This helps use avoid un-needed ones. However, based on how it hasn't been needed thus far, I'm not 100% sure it's worth spec'ing. I would guess it will only be used in web (maybe android/iOS).

All that said, it's obviously easy to implement.

Reply:
This was done in the SDK outside of this PR.